applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <caf/events/ble_common_event.h>
+
+#include "usb_event.h"
+
+#define MODULE dvfs
+#include <caf/events/module_state_event.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_DVFS_LOG_LEVEL);
+
+#include "ld_dvfs_handler.h"
+
+#define DVFS_RETRY_INITIALIZATION_TIMEOUT	K_MSEC(CONFIG_DESKTOP_DVFS_RETRY_INIT_TIMEOUT_MS)
+#define DVFS_RETRY_TIMEOUT			K_MSEC(CONFIG_DESKTOP_DVFS_RETRY_BUSY_TIMEOUT_MS)
+#define DVFS_NUMBER_OF_RETRIES			CONFIG_DESKTOP_DVFS_RETRY_COUNT
+/* Connection intervals used by LLPM are out of Bluetooth LE specification.
+ * The intervals are encoded by OR operation with a magic number of 0x0d00.
+ */
+#define REG_CONN_INTERVAL_LLPM_MASK	0x0d00
+
+static bool in_error_state;
```

Comment:
Consider using
```
enum state {
STATE_DISABLED,
STATE_READY,
STATE_ERROR,
}
```
To improve consistency with other application modules

Reply:
+1 for enums, if not then maybe at least rename it to for example `initialization_err` or something more specific

Reply:
Shouldn't we use module_state states? But I cannot find in api a function which gets current module state

Reply:
Generally, modules use `module_state_event` only to broadcast the current state and use module-specific way to remember their own states. Current approach looks ok.

Reply:
I don't like it. module_state_event  should support both broadcast  and module specific. In that way there won't be a situation where we have different values the one broadcasted and one owned by module

Reply:
but api is not user friendly for such use case - maybe we should create a ticket to fix this problem

Reply:
I am not sure what you mean by "not user friendly" api here. You can simply use `enum module_state` instead of custom state enum and pass it as argument to function that broadcasts module state.

Reply:
Like we lack a way to easy access the state of the module.

Right now we need to have access to a event to check it:

```
static inline bool check_state(const struct module_state_event *event,
		const void *module_id, enum module_state state)
```
		
IMO you shouldn't need to have event to check it. And the name is confusing because we are not checking the current state but state which was sand in the event. The event might been copied by us and is no longer valid